fbpic/lpa_utils/laser/laser_profiles.py

```diff
@@ -693,3 +693,137 @@ def E_field( self, x, y, z, t ):
         See the docstring of LaserProfile.E_field
         """
         return self.summed_profile.E_field( x, y, z, t )
+
+
+class FewCycleLaser( LaserProfile ):
+    """Class that calculates an ultra-short, tightly focussed laser"""
+
+    def __init__( self, a0, waist, tau_fwhm, z0, zf=None, theta_pol=0.,
+                    lambda0=0.8e-6, cep_phase=0., propagation_direction=1 ):
+        """
+        When a laser pulse is so short that it contains **only a few laser cycles**,
+        the standard Gaussian profile :any:`GaussianLaser` is not well-adapted.
+        This is because :any:`GaussianLaser` neglects the fact that different
+        frequencies focus in different ways. In particular, when initializing
+        a :any:`GaussianLaser` (with a short duration :math:`\\tau`) out of
+        focus, the profile at focus will not be the expected one.
+
+        Instead, the :any:`FewCycleLaser` profile overcomes this limitation.
+        The electric field for this profile is given by (see
+        `Caron & Potvilege, Journal of Modern Optics 46, 1881 (1999)
+        <https://www.tandfonline.com/doi/abs/10.1080/09500349908231378>`__):
+
+        .. math::
+
+            E(\\boldsymbol{x},t) = Re\\left[ a_0\\times E_0\,
+            e^{i\phi_{cep}} \\frac{i Z_R}{q(z)}
+            \\left( 1 + \\frac{ik_0}{s}\\left(z-z_0-ct+
+            \\frac{r^2}{2q(z)}\\right)\\right)^{-(s+1)} \\right]
+
+        where :math:`k_0 = 2\pi/\\lambda_0` is the wavevector,
+        :math:`E_0 = m_e c^2 k_0 / q_e` is the field amplitude for :math:`a_0=1`,
+        :math:`Z_R = k_0 w_0^2/2` is the Rayleigh length,
+        :math:`q(z) = z-z_f + iZ_R`, and where :math:`s`
+        controls the duration of the pulse and is given by:
+
+        .. math::
+
+            \omega_0 \\tau_{WFHM} = s\\sqrt{2(4^{1/(s+1)}-1)}
```

Comment:
Just a typo here and a few lines below: FWHM instead of WFHM

Reply:
Good catch, thanks!